src/examples/client/simpleStreamableHttp.ts

```diff
@@ -229,7 +233,10 @@ async function connect(url?: string): Promise<void> {
 
         // Set up elicitation request handler with proper validation
         client.setRequestHandler(ElicitRequestSchema, async request => {
-            console.log('\n🔔 Elicitation Request Received:');
+            if (request.params.mode !== 'form') {
+                throw new McpError(ErrorCode.InvalidParams, `Unsupported elicitation mode: ${request.params.mode}`);
+            }
```

Comment:
This check is technically redundant, since the base client already validates the  `mode` before invoking this handler. However, the request handler is typed to `ElicitRequestSchema` because there is not an `ElicitFormRequestSchema` (there _is_ an `ElicitRequestFormParamsSchema`, but that is an inner shape). So without this code checking for `mode="form"`, TypeScript complains because it is not 100% sure if the payload is of the right type.

I didn't want to boil the ocean and change too much of the contract of `client.setRequestHandler(ElicitRequestSchema...)`, but please give me feedback if you think I should change this further!